### PR TITLE
Introduce cert-manager component

### PR DESCRIPTION
**Pull request by razvan-moj**
**Why**
https://github.com/ministryofjustice/cloud-platform/issues/562

### CODE REVIEW

**Comment by alkar** [comment]
I think we should leave this PR open for ministryofjustice/cloud-platform#587 to build on. Merging right now has no value since we cannot apply on live clusters as-is.

**Comment by alkar** [comment]
Squashed, amended and rebased on top of master.

File: terraform/cloud-platform-components/resources/nginx-ingress/default-certificate.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+kind: Certificate
+metadata:
+  name: default
+  namespace: ingress-controllers
+spec:
+  secretName: default-certificate
+  issuerRef:
+    name: letsencrypt-production
+    kind: ClusterIssuer
+  commonName: '*.apps.cloud-platform-test-8.k8s.integration.dsd.io'
```
Comment (jasonBirchall) [inline terraform/cloud-platform-components/resources/nginx-ingress/default-certificate.yaml]: As discussed. Will need to grab this value from upstream tf state.

File: terraform/cloud-platform-components/resources/nginx-ingress/default-certificate.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  acme:
+    config:
+    - dns01:
+        provider: route53
+      domains:
+      - '*.apps.cloud-platform-test-8.k8s.integration.dsd.io'
```
Comment (jasonBirchall) [inline terraform/cloud-platform-components/resources/nginx-ingress/default-certificate.yaml]: Same as above {cloud-platform-test-8.k8s.integration.dsd.io}

**Comment by alkar** [comment]
@jasonBirchall can you apply this latest commit?

**Pull request merged**